### ISSUE TITLE
Enhancement Requests

### ISSUE THREAD

**Issue by wright12** [equelin/Unity-Powershell#19]
Set-UnityLun - it would be nice if the default was to append to the existing configuration
- or post a warning stating that it will remove the luns from the current hosts. Default "append" would be my choice.

If you could update the title bar to show which array/s you are connected to that would be helpful

Thank you

Please makes sure these boxes are checked before submitting your issue (put an X between the brackets).

- [X ] Check that your Powershell version > 5.0 (Bonus point if you are using the latest version available)
- [X ] Check that you follow the instructions for installing the module

Also, please provide:

- Any instructions that may help to reproduce the problem
- Informations about your host `$PSVerionTable`
- The list of the loaded modules `Get-Module`

Name                           Value
----                           -----
PSVersion                      5.1.14409.1005
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.14409.1005
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1

ModuleType Version    Name                                ExportedCommands
---------- -------    ----                                ----------------
Manifest   3.1.0.0    Microsoft.PowerShell.Management     {Add-Computer, Add-Content, Checkpoint-Computer, Clear-Content...}
Manifest   3.0.0.0    Microsoft.PowerShell.Security       {ConvertFrom-SecureString, ConvertTo-SecureString, Get-Acl, Get-AuthenticodeSignature...}
Manifest   3.1.0.0    Microsoft.PowerShell.Utility        {Add-Member, Add-Type, Clear-Variable, Compare-Object...}
Script     0.12.1     Unity-Powershell                    {Connect-Unity, Disable-UnityFastCache, Disconnect-Unity, Enable-UnityFastCache...}

**Comment by mtboren** [comment]
Hello, @equelin -

About the:

> If you could update the title bar to show which array/s you are connected to that would be helpful

After talking with @wright12 about such things, I added such a tidbit to [commit b018621b6ee7117ffc260fa236a78f7948bf338a](https://github.com/mtboren/Unity-Powershell_fork/commit/b018621b6ee7117ffc260fa236a78f7948bf338a) in the fork on which I worked.  It updates the PowerShell window's title bar with a touch of information about the Unity storage array(s) to which the current session is connected.

To me that is a pretty handy/valuable little enhancement, especially when one is managing several/many Unity arrays.

**Comment by equelin** [comment]
Hi @wright12 , @mtboren 

You're welcome to submit a PR with this feature! ;-)

Concerning the `-Append` I understand your request and I'll think about a safer way to manage hosts. Feel free to provide any ideas.

**Comment by mtboren** [comment]
PR submitted!

As for handling of host access with `Set-UnityLUN`, I feel, like @wright12, that making `-Append` default to `$true` is a possible solution.

Another way that comes to mind is to remove the "overwrite" capability altogether by providing, for host access add/removal for a LUN, parameters `-AddHost` and `-RemoveHost`, so that it is explicit when the user is setting host access for a LUN:  they are either adding or removing (or possibly both in the same call), but there is little [no?] question as to the action that they about to take.  Thoughts on that?

**Comment by mtboren** [comment]
Fab -- new titlebar display functionality!  Though, I see now that I included a bug in the function definition in `Update-TitleBarForUnityConnection.ps1` -- the function would overwrite the PS titlebar altogether with the Unity connection information, instead of appending to whatever the current titlebar was (appending being useful if one has other connection info in their titlebar, like PowerCLI / vSphere connection info, for example).

It was a matter of correcting a variable name in the function.  I have now made a bugfix branch based on the current `develop` branch of https://github.com/equelin/Unity-Powershell, and will submit a PR to rectify the bug that I introduced.  Apologies for the extra work/noise.
